extensions/rick-and-morty/src/characters.tsx

```diff
@@ -0,0 +1,109 @@
+import { Color, Icon, List, Toast, showToast } from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+import { Character, getCharacters } from "rickmortyapi";
+import "cross-fetch/polyfill";
+import { useState } from "react";
+
+export default function Characters() {
+  const [searchName, setSearchName] = useState("");
+
+  const {
+    isLoading,
+    data: characters,
+    pagination,
+  } = useCachedPromise(
+    (name: string) => async (options) => {
+      const response = await getCharacters({ page: options.page, name });
+      if (response.status !== 200) throw new Error(response.statusMessage);
+      const data = response.data.results as Character[];
+      const hasMore = Boolean(response.data.info?.next);
+      return { data, hasMore };
+    },
+    [searchName],
+    {
+      async onWillExecute() {
+        await showToast(Toast.Style.Animated, "Fetching Characters");
+      },
+      async onData(data) {
+        await showToast(Toast.Style.Success, `Fetched ${data.length} Characters`);
+      },
```

Comment:
I don't think this brings any value for the users. Could we skip toasts since there already is a loading indicator 🙂

Reply:
This is the same for all commands 🙂

Reply:
Agreed with your other changes. For the animated toast the only reason I added that is for pagination so one can know that more items have been added. But I'll remove it if it seems redundant.